src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsManifestNegative.cs

```diff
@@ -0,0 +1,212 @@
+                    () => EventSource.GenerateManifest(typeof(Sdt.UnsealedEventSource), string.Empty, strictOptions));
+
+                ExceptionAssert.Throws<ArgumentException>(GetResourceString("EventSource_TypeMustBeSealedOrAbstract"),
+                    () => EventSource.GenerateManifest(typeof(Sdt.UnsealedEventSource), string.Empty, strictOptions));
+
+                // starting with NuGet we allow non-void returning methods as long as they have the [Event] attribute
+                Assert.NotNull(EventSource.GenerateManifest(typeof(Sdt.EventWithReturnEventSource), string.Empty));
+
+                Assert.NotNull(EventSource.GenerateManifest(typeof(Sdt.EventWithReturnEventSource), string.Empty, strictOptions));
+
+                Assert.NotNull(EventSource.GenerateManifest(typeof(Sdt.EventWithReturnEventSource), string.Empty, EventManifestOptions.AllowEventSourceOverride));
+
+                ExceptionAssert.Throws<ArgumentException>(GetResourceString("EventSource_NeedPositiveId", "WriteInteger"),
+                    () => EventSource.GenerateManifest(typeof(Sdt.NegativeEventIdEventSource), string.Empty));
+
+                ExceptionAssert.Throws<ArgumentException>(GetResourceString("EventSource_NeedPositiveId", "WriteInteger"),
+                    () => EventSource.GenerateManifest(typeof(Sdt.NegativeEventIdEventSource), string.Empty, strictOptions));
+
+                ExceptionAssert.Throws<ArgumentException>(GetResourceString("EventSource_NeedPositiveId", "WriteInteger"),
+                    () => EventSource.GenerateManifest(typeof(Sdt.NegativeEventIdEventSource), string.Empty, EventManifestOptions.AllowEventSourceOverride));
+
+                // This check was removed from EventSource
+                // ExceptionAssert.Throws<ArgumentException>(GetResourceString("EventSource_ReservedKeywords", "WriteInteger"),
+                //     () => EventSource.GenerateManifest(typeof(Sdt.OutOfRangeKwdEventSource), string.Empty));
+
+                // ExceptionAssert.Throws<ArgumentException>(
+                //     GetResourceString("EventSource_ReservedKeywords", "WriteInteger"),
+                //     () => EventSource.GenerateManifest(typeof(Sdt.OutOfRangeKwdEventSource), string.Empty));
+
+                ExceptionAssert.Throws<ArgumentException>(
+                    GetResourceString("EventSource_IllegalKeywordsValue", "Kwd1", "0x100000000000"),
+                    GetResourceString("EventSource_KeywordCollision", "Session3", "Kwd1", "0x100000000000"),
+                    // GetResourceString("EventSource_ReservedKeywords", "WriteInteger"),
+                    () => EventSource.GenerateManifest(typeof(Sdt.OutOfRangeKwdEventSource), string.Empty, strictOptions));
+
+#if FEATURE_ADVANCED_MANAGED_ETW_CHANNELS
+            ExceptionAssert.Throws<ArgumentException>(GetResourceString("EventSource_MaxChannelExceeded"),
```

Comment:
Same question about these compilation constants, who sets them, and when.  Also, nit, indentation here and below looks off.